<a href="https://colab.research.google.com/github/Tiagee0730/Prediction-of-Product-Sales/blob/main/Prediction%20of%20Product%20Sales%201.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Prediction of Product Sales
- Tia Gee

## Project Overview

## Load and Inspect Data

### Loading Data

1. import pandas

In [1]:
import pandas as pd

2. mount the drive

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [5]:
sales_predicitons = '/content/sales_predictions.csv'

use pandas to read the sales predictions

In [6]:
df = pd.read_csv(sales_predicitons)

use df.info() and df.head() to preview the first rows and a summary of your DataFrame's column

In [8]:
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


## Clean Data
### Data Cleaning

In [10]:
df.shape

(8523, 12)

1. How many rows and columns?
there are 8523 rows and 12 columns

2. What are the datatypes of each variable?

In [11]:
df.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

3. Are there duplicates? If so, drop them. **there are not**

In [13]:
df.duplicated().sum()

0

4. Identify missing values

In [15]:
df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [16]:
df.dropna(subset=['Item_Weight', 'Outlet_Size'], inplace=True)
df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

5. I dropped the missing rows because it was such a large number it would skew the data. 

6. I confirmed there is no missing data by using df.isna().sum() after to see.
7. find and fix any inconsistent categories of data **Could not find any**

In [17]:
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15        9.300          Low Fat         0.016047   
1           DRC01        5.920          Regular         0.019278   
2           FDN15       17.500          Low Fat         0.016760   
4           NCD19        8.930          Low Fat         0.000000   
5           FDP36       10.395          Regular         0.000000   

      Item_Type  Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
0         Dairy  249.8092            OUT049                       1999   
1   Soft Drinks   48.2692            OUT018                       2009   
2          Meat  141.6180            OUT049                       1999   
4     Household   53.8614            OUT013                       1987   
5  Baking Goods   51.4008            OUT018                       2009   

  Outlet_Size Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0      Medium               Tier 1  Supermarket Type1          3735.1380  
1      Medium               Tier 3  Supermarket Type2           443.4228  
2      Medium               Tier 1  Supermarket Type1          2097.2700  
4        High               Tier 3  Supermarket Type1           994.7052  
5      Medium               Tier 3  Supermarket Type2           556.6088

there are no inconsistencies in the title

In [26]:
df.nunique()

Item_Fat_Content                5
Item_Visibility              4280
Item_Type                      16
Item_MRP                     3806
Outlet_Identifier               5
Outlet_Establishment_Year       5
Outlet_Size                     3
Outlet_Location_Type            3
Outlet_Type                     2
Item_Outlet_Sales            2449
dtype: int64

I don't think item weight, item identifier, outlet identifier has anything to do with product sales so I am going to drop that column.

In [35]:
df = df.drop(columns = 'Outlet_Identifier')

Lets see if the data types are correct **they are**

*   List item
*   List item



In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4650 entries, 0 to 8522
Data columns (total 11 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            4650 non-null   object 
 1   Item_Fat_Content           4650 non-null   object 
 2   Item_Visibility            4650 non-null   float64
 3   Item_Type                  4650 non-null   object 
 4   Item_MRP                   4650 non-null   float64
 5   Outlet_Identifier          4650 non-null   object 
 6   Outlet_Establishment_Year  4650 non-null   int64  
 7   Outlet_Size                4650 non-null   object 
 8   Outlet_Location_Type       4650 non-null   object 
 9   Outlet_Type                4650 non-null   object 
 10  Item_Outlet_Sales          4650 non-null   float64
dtypes: float64(3), int64(1), object(7)
memory usage: 435.9+ KB


In [22]:
df.head()

Item_Identifier Item_Fat_Content  Item_Visibility     Item_Type  Item_MRP  \
0           FDA15          Low Fat         0.016047         Dairy  249.8092   
1           DRC01          Regular         0.019278   Soft Drinks   48.2692   
2           FDN15          Low Fat         0.016760          Meat  141.6180   
4           NCD19          Low Fat         0.000000     Household   53.8614   
5           FDP36          Regular         0.000000  Baking Goods   51.4008   

  Outlet_Identifier  Outlet_Establishment_Year Outlet_Size  \
0            OUT049                       1999      Medium   
1            OUT018                       2009      Medium   
2            OUT049                       1999      Medium   
4            OUT013                       1987        High   
5            OUT018                       2009      Medium   

  Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0               Tier 1  Supermarket Type1          3735.1380  
1               Tier 3  Supermarket Type2           443.4228  
2               Tier 1  Supermarket Type1          2097.2700  
4               Tier 3  Supermarket Type1           994.7052  
5               Tier 3  Supermarket Type2           556.6088

check for inconsistencies in categorical values

In [31]:
df['Outlet_Establishment_Year'].value_counts

<bound method IndexOpsMixin.value_counts of 0       1999
1       2009
2       1999
4       1987
5       2009
        ... 
8517    1997
8518    1987
8520    2004
8521    2009
8522    1997
Name: Outlet_Establishment_Year, Length: 4650, dtype: int64>

8. For any numerical columns, obtain the summary statisitics of each (min, max, mean)

In [34]:
df.describe().round(4)

Item_Visibility   Item_MRP  Outlet_Establishment_Year  \
count        4650.0000  4650.0000                  4650.0000   
mean            0.0607   141.7163                  1999.1905   
std             0.0446    62.4205                     7.3888   
min             0.0000    31.4900                  1987.0000   
25%             0.0260    94.4094                  1997.0000   
50%             0.0497   142.9799                  1999.0000   
75%             0.0887   186.6142                  2004.0000   
max             0.1883   266.8884                  2009.0000   

       Item_Outlet_Sales  
count          4650.0000  
mean           2272.0375  
std            1497.9647  
min              69.2432  
25%            1125.2020  
50%            1939.8083  
75%            3111.6163  
max           10256.6490

## Exploratory Data Analysis

## Explanatory Data Analysis